packages/components/ovh-shell/README.md

```diff
@@ -1 +1,178 @@
-# ovh-shell
+# @ovh-ux/shell
+
+The goal of this lib is to provide utilities to communicate with a Shell. It exposes an API for that purpose, it also exposes a list of Plugins you can use with the Shell.
+
+## Table of contents
+
+- [Lib usage](#usage)
+- [API Docs](#api)
+  - [Shell methods](#shell)
+    - [initShell](#initShell)
+  - [Plugins API](#plugins)
+    - [i18n](#i18n)
+    - [routing](#routing)
+    - [auth](#auth)
+    - [ux](#ux)
+    - [tracking](#tracking)
+    - [navigation](#navigation)
+
+## Usage
+
+To use this lib, you have to add `@ovh-ux/shell` to your dependencies and then build the package under `packages/components/ovh-shell/`.
+
+The lib is fully tree-shaked, so to import something you just have to import it as an esModule :
+
+```ts
+import { shell } from '@ovh-ux/shell';
+```
+
+## API
+
+### Shell
+
+#### `initShell`
+
+This method initializes a Shell given an iframe. You can initiate a Shell without an iframe, but keep in mind that the communications will use the `window` object instead of the `iframe.contentWindow` object.
+
+_Example :_
+
+```ts
+import { shell } from '@ovh-ux/shell';
+
+const shellApi = shell.initShell(yourIframe);
+```
+
+This method inits a shell and returns an API that allows you to communicate with it.
+
+The returned object contains the following methods :
+
+- `emitEvent`(`(eventId: string, data: unknown) => void`) : This method allows to emit a custom event from the Shell.
+  - `eventId`: The `name`/`id` of your event.
+  - `data`: Contains the data you want to send with your event. It can be everything that can be parsed by the navigator, objects such as callbacks or Promises won't work.
+
+- `registerPlugin`(`(pluginId: string, pluginApi: Record<string, Callback>) => void`): Register a plugin by giving its id and API.
+  - `pluginId`: Your plugin `name`/`id`.
```

Comment:
Should all plugin name/id be unique? We should have a list of plugins reserved by the shell.

Maybe it would be good practice to register plugin id with the app as prefix (e.g. `my-app:my-plugin`) to avoid errors (both for projects using the Shell that could try to redefine an existing plugin and for the shell that could have new plugins and redefine a plugin with a name that another project is already using).